### ISSUE TITLE
Insufficient colour contrast

### ISSUE THREAD

**Issue by selfthinker** [splitbrain/dokuwiki#3443]
Most text needs to have a colour contrast of 4.5:1 with the background (unless it is large text, inactive or part of a logo). And non-text of interface components and important images need to have a contrast of 3:1 with adjacent colours. That is so that people with visual impairments can read the text well and see important non-text content well.

These are most instances where that minimum colour contrast is not met.
They either fail [1.4.3 Contrast (Minimum)](https://www.w3.org/TR/WCAG21/#contrast-minimum) or [1.4.11 Non-text Contrast](https://www.w3.org/TR/WCAG21/#non-text-contrast).

### General template (every page)

* [ ] The **page tool icons** have an insufficient colour contrast of 1.5:1 against the background (`#cccccc` on `#fbfaf9`). This is a fail of 1.4.11 and a high priority as it could mean that some people won't be able to see and therefore use these important functionalities at all.
![](https://user-images.githubusercontent.com/108893/111907509-9319f980-8a4d-11eb-8542-15cb46043aa9.png)

* [ ] The **page tools links** have an insufficient colour contrast of 4.27:1 when they are focused or hovered (`#2b73b7` on `#eeeeee`). This is a fail of 1.4.3 and a low priority.
![](https://user-images.githubusercontent.com/108893/111907529-adec6e00-8a4d-11eb-9b8b-168460f87243.png)

* [ ] **Buttons and input fields** (including checkboxes) have a border colour of `#cccccc`, usually against `#ffffff`. That is a colour contrast of 1.6:1 and is insufficient. When they are focused or active it's `#999999` against `#ffffff` which is also insufficient with a contrast ratio of 2.8:1. This fails 1.4.11 and is a medium priority as inputs are very important for interactions in general.
![](https://user-images.githubusercontent.com/108893/111905767-43373480-8a45-11eb-8a0b-d48c2b5305a9.png)

* [ ] The border of the **search box** has insufficient colour contrast with the surrounding background. The background is a gradient, so taking the lightest of those colours (`#fbfAf9`) against the border colour (`#cccccc`) means the contrast ratio is 1.5:1. The colours in the input's active and focused states are also insufficient (`#999999` against `#ffffff` is a contrast ratio of 2.8:1). This fails 1.4.11 and has a medium priority.
![](https://user-images.githubusercontent.com/108893/111905201-5eed0b80-8a42-11eb-9281-b09ee2072967.png)

* [ ] The **page ID** tab on the top right of the main content has a low colour contrast ratio of 2.85:1 (`#999999` on `#ffffff`). This fails 1.4.3 and has a low priority.
![](https://user-images.githubusercontent.com/108893/111905246-8e037d00-8a42-11eb-8809-89687afc88cb.png)

### [Edit view](https://grimm.haz.wiki/animals?do=edit)

* [ ] The **draft message** ("Draft autosaved on [datetime]") has an insufficient colour contrast of 2.85:1 (`#999999` on `#ffffff`). This fails 1.4.3 and has a low priority.
![](https://user-images.githubusercontent.com/108893/111905853-b17bf700-8a45-11eb-8385-3aac02461380.png)

* [ ] The text (`#666666`) and links (`#008800`) on even backgrounds (`#dddddd`) in the **link wizard** have an insufficient colour contrast of 4.2:1 and 3.4:1. This fails 1.4.3.
![](https://user-images.githubusercontent.com/108893/111905887-dec8a500-8a45-11eb-902d-03a13639c6ad.png)

### [Search results](https://grimm.haz.wiki/start?do=search&id=start&sf=1&q=king)

* [ ] The colour contrast of the **search snippets** is an insufficient 2.85:1 (`#999999` on `#ffffff`). This fails 1.4.3.
![](https://user-images.githubusercontent.com/108893/111905948-27805e00-8a46-11eb-8bd6-61f307b46ae6.png)

### [Syntax page](https://grimm.haz.wiki/wiki:syntax)

* [ ] The open/close indicator icon of the table of contents and the sidebar in mobile view is between `#cccccc` and `#eeeeee` against `#ffffff`. The colour contrast between those is an insufficient 1.16:1. This fails 1.4.11 and has a medium priority as there is nothing else that visually shows it has an open/close functionality.
![](https://user-images.githubusercontent.com/108893/111906000-729a7100-8a46-11eb-8286-13788fac1f10.png)

## Other non-fail accessibility issues

These don't strictly fail WCAG but are still accessibility issues. Fixing them is recommended but optional for passing WCAG.

* [ ] The text and border on the footer badges have insufficient colour contrast. When they are focused or active, the text contrast is still not good but the border contrast is fine. They could count as logos, so would be exempt. But the reduced opacity makes it worse.
![](https://user-images.githubusercontent.com/108893/111908141-fefd6180-8a4f-11eb-89a7-4e45355796e0.png)

* [ ] The **page tool icons** are barely visible when changing colours in Firefox to a light colour scheme and would be invisible if the background colour was changed to the same colour as the icons (`#cccccc`). This has a higher priority as it could mean that some people won't be able to see them at all.
![](https://user-images.githubusercontent.com/108893/111908041-9910da00-8a4f-11eb-85ef-398dedb27aad.png)

## How to fix it

In many cases the fix is to make the text colour a bit darker. But especially for fails of 1.4.11 there can be other fixes.
For example, in the case of the inputs' border colour it's not necessarily the border that needs to be fixed. It's important that whatever makes it stand out as a button/input needs to have a good colour contrast. That could also be done with the background colour or something else.
